include/envoy/server/options.h

```diff
@@ -16,30 +16,43 @@ namespace Envoy {
 namespace Server {
 
 /**
- * Whether to run Envoy in serving mode, or in config validation mode at one of two levels (in which
- * case we'll verify the configuration file is valid, print any errors, and exit without serving.)
+ * What mode to run Envoy in: serving, or validation with various levels of fidelity. In validation
+ * mode, Envoy will check that the configuration file is valid (with different modes verifying to
+ * different degrees), print any errors, and exit without serving.
  */
 enum class Mode {
   /**
    * Default mode: Regular Envoy serving process. Configs are validated in the normal course of
    * initialization, but if all is well we proceed to serve traffic.
    */
-  Serve,
+  Serve = 3,
 
   /**
-   * Validate as much as possible without opening network connections upstream or downstream.
+   * High fidelity validation: completely load and initialize the config, and then exit without
+   * running the listener loop.
+   *
+   * This provides the highest level of validation since it initializes Envoy completely without
+   * serving any traffic.
    */
-  Validate,
+  InitOnly = 2,
 
   /**
-   * Completely load and initialize the config, and then exit without running the listener loop.
+   * Medium fidelity validation: load and initialize with the configuration, assuming full access to
+   * the file system but without opening network connections, and then exit.
```

Comment:
Perhaps to make it abundantly clear, mention that this doesn't validate dynamic config, perhaps something like: `without opening network connections and validating dynamic configuration`?